balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -105,6 +116,12 @@ func (pickfirstBuilder) ParseConfig(js json.RawMessage) (serviceconfig.LoadBalan
 	return cfg, nil
 }
 
+// EnableHealthListener updates the state to configure pickfirst for using a
+// generic health listener.
+func EnableHealthListener(attrs *attributes.Attributes) *attributes.Attributes {
+	return attrs.WithValue(enableHealthListenerKeyType{}, enableHealthListenerValue)
+}
```

Comment:
For these types of functions we prefer to make them operate on the thing that contains the attributes.  In this case, that would be a `resolver.State`.

Reply:
Changed the function to accept `resolver.State`.